data/beaconrestapi/src/main/java/tech/pegasys/teku/beaconrestapi/handlers/v1/debug/GetDeposits.java

```diff
@@ -0,0 +1,130 @@
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import io.javalin.http.Context;
+import io.javalin.plugin.openapi.annotations.HttpMethod;
+import io.javalin.plugin.openapi.annotations.OpenApi;
+import io.javalin.plugin.openapi.annotations.OpenApiContent;
+import io.javalin.plugin.openapi.annotations.OpenApiResponse;
+import java.util.List;
+import java.util.function.Function;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.jetbrains.annotations.NotNull;
+import tech.pegasys.teku.api.ChainDataProvider;
+import tech.pegasys.teku.api.DataProvider;
+import tech.pegasys.teku.api.response.v1.debug.GetDepositsResponse;
+import tech.pegasys.teku.beaconrestapi.MigratingEndpointAdapter;
+import tech.pegasys.teku.infrastructure.json.types.SerializableTypeDefinition;
+import tech.pegasys.teku.infrastructure.restapi.endpoints.AsyncApiResponse;
+import tech.pegasys.teku.infrastructure.restapi.endpoints.EndpointMetadata;
+import tech.pegasys.teku.infrastructure.restapi.endpoints.RestApiRequest;
+import tech.pegasys.teku.spec.Spec;
+import tech.pegasys.teku.spec.datastructures.operations.Deposit;
+import tech.pegasys.teku.validator.coordinator.DepositProvider;
+
+@SuppressWarnings("unused")
+public class GetDeposits extends MigratingEndpointAdapter {
+
+  private static final Logger LOG = LogManager.getLogger();
+  public static final String ROUTE = "/eth/v1/debug/beacon/deposits";
+  private static final String DEFAULT_STATE_ID = "head";
+
+  private static final SerializableTypeDefinition<List<Deposit>> DEPOSITS_RESPONSE_TYPE =
+      SerializableTypeDefinition.<List<Deposit>>object()
+          .name("GetDepositsResponse")
+          .withField(
+              "data", listOf(Deposit.SSZ_SCHEMA.getJsonTypeDefinition()), Function.identity())
+          .build();
+
+  private final ChainDataProvider chainDataProvider;
+  private final DepositProvider depositProvider;
+  private final Spec spec;
+
+  public GetDeposits(
+      final DataProvider provider, final DepositProvider depositProvider, final Spec spec) {
+    this(provider.getChainDataProvider(), depositProvider, spec);
+  }
+
+  public GetDeposits(
+      final ChainDataProvider chainDataProvider,
+      final DepositProvider depositProvider,
+      final Spec spec) {
+    super(
+        EndpointMetadata.get(ROUTE)
+            .operationId("getDeposits")
+            .summary("Deposits not yet included on the chain")
+            .description(
+                "Range of deposits that could be used in new block if we are assigned to be a proposer.")
+            .tags(TAG_DEBUG)
+            .response(SC_OK, "Request successful", DEPOSITS_RESPONSE_TYPE)
+            .build());
+    this.chainDataProvider = chainDataProvider;
+    this.depositProvider = depositProvider;
+    this.spec = spec;
+  }
+
+  @OpenApi(
+      path = ROUTE,
+      method = HttpMethod.GET,
+      summary = "Deposits not yet included on the chain",
```

Comment:
summary needs to be super short, it is used to display in a menu on our api docs here: https://consensys.github.io/teku
eg.  would be `/eth/v1/beacon/pool/attestations` which potentially is similar to this endpoint (in that it gets things from the attestations pool) - its description is 'Get Attestations', and a block selector one is 'Get Block attestations'.

Maybe 'Get deposits' or 'Get pending deposits'?

Reply:
Looking at @ajsutton comment, it should be "Get deposits"